src/cpp/shared_core/system/PosixSystem.cpp

```diff
@@ -43,13 +43,24 @@ namespace posix {
 
 namespace {
 
+// Apple uses int for gid_t whereas unix uses uint32
+#ifndef __APPLE__
+   typedef gid_t GIDTYPE;
+#else
+   typedef int GIDTYPE;
```

Comment:
I think you could also bring in `<unistd.h>` to get `gid_t`, which woudl be slightly cleaner? I believe we do this in other source files using `gid_t`.

Reply:
Ah, but that doesn't help because ::getgrouplist takes an `int*`, as per https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man2/getgid.2.html.